src/observer.ts

```diff
@@ -40,14 +44,31 @@ class PerformanceObserver implements PollingPerformanceObserver {
   }
 
   public observe(options?: PerformanceObserverInit): void {
-    if (!options || !options.entryTypes) {
+    if (!options || (!options.entryTypes && !options.type)) {
       throw new Error(ERRORS["no-entry-types"]);
     }
 
-    const entryTypes: string[] = options.entryTypes.filter(isValidType);
+    if (options.entryTypes && options.type) {
+      throw new Error(ERRORS["both-entry-types"]);
+    }
+
+    let rawEntryTypes: string[] = [];
```

Comment:
Could this code fragment be rewritten? This way we can eliminate redundant array initialization `rawEntryTypes: string[] = []`

```ts
  public observe(options?: PerformanceObserverInit): void {
    if (!options) {
      throw new Error(ERRORS["no-entry-types"]);
    }

    let rawEntryTypes: string[];
    if (options.entryTypes && options.type) {
      throw new Error(ERRORS["both-entry-types"]);
    } else if (options.entryTypes) {
      rawEntryTypes = options.entryTypes;
    } else if (options.type) {
      rawEntryTypes = [options.type];
    } else {
      throw new Error(ERRORS["no-entry-types"]);
    }
```

Reply:
Have cleaned this up in https://github.com/fastly/performance-observer-polyfill/commit/6d7f6c77b4eae56ce970682839657ba4af07035f